### PR TITLE
Added support for application/yaml response payloads.

### PR DESCRIPTION
**Pull request by dblock**
### Description

On top of https://github.com/opensearch-project/opensearch-api-specification/pull/360, Added application/yaml support and responses to cat APIs. This one requires deserializing the response body into an object to validate.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: spec/schemas/cat.indices.yaml [hunk 1]
```diff
@@ -29,13 +29,13 @@ components:
           type: string
         docs.count:
           description: available docs
-          oneOf:
+          anyOf:
             - type: string
             - nullable: true
               type: string
```
Comment (nhtruong) [inline spec/schemas/cat.indices.yaml:35]: probably scope creeping but `nullable` is not a proper JSON schema keyword. Even if it was, this is redundant because nullable string already encapsulates non-nullable string.

In this case and a few below, we can repalce `anyOf: ...` with `type: [string, null]` according to [this](https://json-schema.org/understanding-json-schema/reference/type)

Comment (dblock) [reply spec/schemas/cat.indices.yaml:35]: I tried this but it failed with some other error. Care to open an issue with some details to replace?

**Review by nhtruong — DISMISSED** [review]

**Review by nhtruong — APPROVED** [review]

**Review by nhtruong — APPROVED** [review]

**Comment by nhtruong** [comment]
This is stuck at "Checking for ability to merge automatically…"
